### ISSUE TITLE
[terra-date-picker] date format no longer accepted

### ISSUE THREAD

**Issue by hboege** [cerner/terra-framework#874]
# Bug Report
terra-date-picker no longer sets the date with previously working value.

## Description
Previously, I set the value on a DatePicker to 20190801, which would display as 08/01/2019 (in en-US) in the date picker.
With the 4.12.0 changes, it now displays as 20190801.   While this might be a necessary change, this is not passive.

## Steps to Reproduce
1.  Create a DatePicker using < 4.12.0, passing in a value YYYYMMDD.  Verify the appearance.
2.  Upgrade to 4.12.0 without changing the code.  The date doesn't look right anymore.

## Additional Context / Screenshots
Dates using < 4.12.0
![formatted date](https://user-images.githubusercontent.com/30238809/64461992-9399ac00-d0c4-11e9-9e03-fccd1cfc86c4.png)

Dates using 4.12.0
![unformatted date](https://user-images.githubusercontent.com/30238809/64462016-a4e2b880-d0c4-11e9-95dd-39d0651f4da5.png)

## Possible Solution
Revert the changes to fix the minor version.  Then add the changes back in for a major version release. 

## Environment

* Component Name and Version: terra-date-picker  4.12.0
* Browser Name and Version: IE and Chrome
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile):

**Comment by nramamurth** [comment]
I believe that this goes back to the [discussion](https://github.com/cerner/terra-framework/pull/851#discussion_r316253748) that this is a bug fix (meaning that this has corrected a wrong behavior that existed) and not a breaking change per se. I could be wrong, tho. 

cc @benbcai @bjankord

**Comment by benbcai** [comment]
I agree with @nramamurth 's comment above. Prior to 4.12.0, the _controlled_ date-picker would accept ordinal dates (e.g. `01011234`; `20190801`) as valid values and converts it to a readable, formatted date as described on issue #800. The fix for this on PR #851 is to prevent ordinal date (as they are confusing to read) as valid values and only accepts values that must be either in the more strict `YYYY-MM-DD` ISO-8601 format or the locale's format (e.g. MM/DD/YYYY for U.S.) as described on the [4.12.0 release change log](https://engineering.cerner.com/terra-ui/components/terra-date-picker/date-picker/change-log).

I would agree that this is technically a non-passive change but the change is to address inputs that the component considers misformed.

@BenBoersma @nramamurth @bjankord @hboege

**Comment by hboege** [comment]
I'm all for fixing problem code.  Its just that it broke working functionality.  If as consumers, we are in the middle of releasing our components/applications, having these changes in a minor version change means a component or application could get released broken and not know it.

**Comment by ryanthemanuel** [comment]
I argue that the working functionality is not correct.  The control is broken as is and could get into a situation where bad dates are entered as values.  We don't want to get into the habit of making people have to major version bump in order to get fixes for defective components.
